### PR TITLE
Add type annotations for Rembert/Splinter and copies

### PR DESCRIPTION
**Pull request by jacobdineen**
Splinter and Rembert (Torch) type annotations.
Dependencies on Bert (Torch).
Raw files w/o running make fixup or make fix-copies

#16059
@Rocketknight1

### CODE REVIEW

File: src/transformers/models/bert/modeling_bert.py
Comment (Rocketknight1) [inline src/transformers/models/bert/modeling_bert.py:129]: Hey, I still see a lot of formatting changes in this PR that will probably propagate through the codebase when we `make fix-copies`! Do you have something like a save hook for `PyCharm` that always runs `black`?

Comment (Rocketknight1) [reply src/transformers/models/bert/modeling_bert.py]: This looks much better, thank you!

**Comment by Rocketknight1** [comment]
By the way, GitHub is having some issues right now and tests may not be running. Feel free to continue with the PR, but we'll need to wait for them to come back before we can merge!

**Comment by Rocketknight1** [comment]
Tests are back, going to try fixing copies etc. now, one sec!

**Comment by jacobdineen** [comment]
Is this a private file? This was the main error I was receiving after run fix-copies

```
self = <module 'transformers.models.template_bi' from '/home/runner/work/transformers/transformers/src/transformers/models/template_bi/__init__.py'>
module_name = 'modeling_template_bi'

    def _get_module(self, module_name: str):
        try:
            return importlib.import_module("." + module_name, self.__name__)
        except Exception as e:
            raise RuntimeError(
                f"Failed to import {self.__name__}.{module_name} because of the following error (look up to see its traceback):\n{e}"
>           ) from e
E           RuntimeError: Failed to import transformers.models.template_bi.modeling_template_bi because of the following error (look up to see its traceback):
E           name 'Optional' is not defined```

**Comment by jacobdineen** [comment]
I see. Definitely scratched my head at these for a while - Thanks so much for your help!

**Review by Rocketknight1 — APPROVED** [review]
Tests are all passing now! Thanks again for your work on this, and your patience as we figured out the templating issues!

**Pull request merged**
